lib/philomena_web/plugs/content_security_policy_plug.ex

```diff
@@ -66,10 +66,31 @@ defmodule PhilomenaWeb.ContentSecurityPolicyPlug do
   defp cdn_uri, do: Application.get_env(:philomena, :cdn_host) |> to_uri()
   defp camo_uri, do: Application.get_env(:philomena, :camo_host) |> to_uri()
 
-  defp default_script_src, do: vite_hmr?(do: "'self' localhost:5173", else: "'self'")
+  # Use the "current host" in vite HMR mode for whatever the "current host" is.
+  # Usually it's `localhost`, but it may be some other private IP address, that
+  # you use to test the frontend on a mobile device connected via a local Wi-Fi.
+  defp default_script_src(host) do
+    # Workaround for a compile warning where `host` variable is unused if we
+    # inline the if branches into the `vite_hmr?` macro.
+    is_vite_hmr = vite_hmr?(do: true, else: false)
+
+    if is_vite_hmr do
+      "'self' #{host}:5173"
+    else
+      "'self'"
+    end
+  end
+
+  defp default_connect_src(host) do
+    # Same workaround as in `default_script_src/1`
+    is_vite_hmr = vite_hmr?(do: true, else: false)
```

Comment:
same idea here

Reply:
ideally fold both functions into the same vite_hmr? do ... else ... end construct